### PR TITLE
Milestone 4 (excluding temperature)

### LINKED ISSUE #38: Fail gracefully when JSON message body is empty
By empty I mean truly empty, not `{}`.

### LINKED ISSUE #51: Update instructions to not be a queue

### LINKED ISSUE #53: Deleting delivery does not clear token
Deleting a delivery does not clear a token when the delivery is awaiting authentication.

### LINKED ISSUE #55: Enforce proper state transitions
Forbid state transitions that aren't legal, e.g.: (`IN_QUEUE` -> `MOVING_TO_DESTINATION`)

### LINKED ISSUE #58: Write tests to ensure lock is being triggered correctly across state transitions

### PR DESCRIPTION
**Pull request by ltorroba**
Applies core changes required for milestone 4. Temperature sensor endpoint not included in this PR (we still need to discuss the best way to implement the feature).

## Routes
- [x] /robot/{id}/batch
- [x] /robot/{id}/correction
- [x] /robot/{id}/distance
- [x] /robot/{id}/motor
- [x] /robot/{id}/angle
- [x] /robot/{id}/verify
- [x] /delivery/{id} [PATCH]

## Issues
- [x] Close #51 
- [x] Close #53

### CODE REVIEW

File: flaskapp/flaskapp.py [hunk 1]
```diff
@@ -26,7 +26,8 @@ def get_cache():
     It can store arbitrary objects.
     """
     if not hasattr(g, 'cache'):
-        g.cache = shelve.open(app.config['SHELVE_FILENAME'], protocol=2)
+        g.cache = shelve.open(app.config['SHELVE_FILENAME'], protocol=2,
+                              writeback=True)
```
Comment (james-od) [inline flaskapp/flaskapp.py:29]: what does this do?

File: flaskapp/flaskapp.py [hunk 2]
```diff
@@ -406,149 +411,177 @@ def target_delete(id):
 #                                         #
 #              ROBOT ROUTES               #
 #                                         #
+def get_robot(id):
+    if 'robots' not in get_cache():
```
Comment (james-od) [inline flaskapp/flaskapp.py:433]: slightly confused by this?

**Review by james-od — COMMENTED** [review]
Looks good, I've got a couple of questions but should be fine :+1:

**Comment by ltorroba** [comment]
@james-od Fixed the bug, it was because of what we talked about. The bare minimum for milestone 4 (apart from the thermometer) should be done.

**Review by james-od — APPROVED** [review]
Nice

**Pull request merged**